src/vision/status_quo/alan_iteratively_regresses.md

```diff
@@ -0,0 +1,113 @@
+
+However, there remains a significant performance delta between the Java version and the Rust version. Alan is not sure his Rust-rewrite attempt is going to get anywhere beyond the prototype stage.
+
+## 🤔 Frequently Asked Questions
+
+*Here are some standard FAQ to get you started. Feel free to add more!*
+
+### **What are the morals of the story?**
+
+1. Rust promises great performance, but when performance is not meeting one's targets, it is hard to know what to do next. Rust mostly leans on leveraging existing tools for native code development, but those tools are (a.) foreign to many of our developers, (b.) do not always measure up to what our developers have access to elsewhere, (c.) do not integrate as well with Rust as they might with C or C++.
+
+2. Lack of certain language features leads developers to use constructs like `#[async_trait]` which add performance overhead that is (a.) hard to understand and (b.) may be significant.
+
+3. Rust makes some things very explicit, e.g. the distinction between `Box<T>` versus `T` is quite prominent. But Rust's expressive type system also makes it easy to compose types without realizing how large they have gotten.
+
+4. Programmers do not always have a good mental model for where expensive moves are coming from.
+
+5. An important specific instance of (1c.) for the async vision: Native code tools do not have any insight into Rust's async model, as that is even more distant from the execution model of C and C++.
+
+6. We can actually generalize (5.) further: When async performance does not match expectations, developers do not have much insight into whether the performance pitfalls arise from issues deep in the async executor that they have selected, or if the problems come directly from overheads built into the code they themselves have written.
+
+### **What are the sources for this story?**
+
+Discussions with engineers at Amazon Web Services.
+
+### **Why did you choose Alan to tell this story?**
+
+I chose Alan because he is used to Java, where these issues play out differently.
+
+Java has very mature tooling, including for performance investigations. Alan is frustrated by his attempts to use (or even identify) equivalent tools for Rust.
+
+With respect to memory traffic: In Java, every object is handled via a reference, and those references are cheap to copy. (One pays for that convenience in other ways, of course.)
+
+
+### **How would this story have played out differently for the other characters?**
+
+From her C and C++ background, [Grace][] probably would avoid letting her types get so large. But then again, C and C++ do not have enums with a payload, so even Grace may have fallen in the same trap that Alan did (of assuming that the cost of moving an enum value is proportional to its current variant, rather than to its type's overall size).
```

Comment:
Grace is probably likely to give more informed complaints or at least feedback on how the Rust perf experience compares to C/C++ perf experience. Eg she might report that her experience with gcc based projects have had very useful debuginfo that worked well with perf, while clang-based projects had weaker support there.

Reply:
```suggestion
From her C and C++ background, [Grace][] probably would avoid letting her types get so large. But then again, C and C++ do not have enums with a payload, so Grace would likely have fallen in the same trap that Alan did (of assuming that the cost of moving an enum value is proportional to its current variant, rather than to its type's overall size).
```

I definitely think Grace would fall into the same trap. I think most programmers don't think about this so actively.

Reply:
I guess my main point is that if any of our non-Barbara is going to sidestep this mistake, it would be Grace. But I don't object to your suggest change (which I think is orthogonal to the note I put about about whether Grace might have more infomed complaints about perf with Rust vs perf with C/C++).

Reply:
addressed the two points in 5b656fc3c9b0d7ceda09fd75135519bca021fc66 and 250646b0a7334f5a5e849c3a5ac36bf31eed9114, respectively. (Decided to just take niko's suggestion as written, rather than quibble about what Grace might have done.)